### DISCUSSION TITLE
pyenv is not installed on the renovate-full container

### DISCUSSION THREAD

**Discussion by stan-stately** [renovatebot/renovate#26649] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.126.1-full

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

When i run renovate and it tries to recreate my pipenv lockfile I get the error:
```
DEBUG: pipenv.updateArtifacts(<REDACTED>/Pipfile) (repository=<REDACTED>, branch=renovate-action/all)
DEBUG: Executing command (repository=<REDACTED>, branch=renovate-action/all)
       "command": "pipenv lock"
DEBUG: rawExec err (repository=<REDACTED>, branch=renovate-action/all)
       "err": {
         "cmd": "/bin/sh -c pipenv lock",
         "stderr": "Warning: Python 3.11.5 was not found on your system...\nNeither 'pyenv' nor 'asdf' could be found to install Python.\nYou can specify specific versions of Python with:\n$ pipenv --python path/to/python\n",
```

@viceice mentions [here](https://github.com/renovatebot/renovate/discussions/14393#discussioncomment-4785535) that the python version should be downloaded on the fly but I dont think thats possible because pyenv is not installed.

This is not an error with my setup. I can see in the [renovate dockerfile](https://github.com/renovatebot/renovate/blob/de68769fc241cf15afc32ae4eddc264194dedc48/lib/modules/manager/custom/regex/__fixtures__/Dockerfile#L138) that pyenv is not installed anywhere.

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: pipenv.updateArtifacts(<REDACTED>/Pipfile) (repository=<REDACTED>, branch=renovate-action/all)
DEBUG: Executing command (repository=<REDACTED>, branch=renovate-action/all)
       "command": "pipenv lock"
DEBUG: rawExec err (repository=<REDACTED>, branch=renovate-action/all)
       "err": {
         "cmd": "/bin/sh -c pipenv lock",
         "stderr": "Warning: Python 3.11.5 was not found on your system...\nNeither 'pyenv' nor 'asdf' could be found to install Python.\nYou can specify specific versions of Python with:\n$ pipenv --python path/to/python\n",
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by stan-stately** [comment]
This issue can be fixed by installing pyenv during your dockerfile init:
```
curl https://pyenv.run | bash
export PYENV_ROOT="$HOME/.pyenv"
[[ -d $PYENV_ROOT/bin ]] && export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init -)"
```

I'm leaving this here as a reference for anyone else who runs into the issue. I wont be making a minimal reproduction

**Reply by viceice** [reply]
Wrong! Set `RENOVATE_BINARY_SOURCE=install` in env and you get the required python version installed on the fly without pyenv.

**Comment by rarkins** [comment, marked as answer]
If you want dynamic installation of python etc then you should set binarySource=install

**Comment by rarkins** [comment]
This is not a bug, closing
